tools/automate_gcp_os_version.py

```diff
@@ -0,0 +1,118 @@
+        print("Exception occurred: ", ex)
+
+
+def compare_versions(recent_os, line):
+    """
+    Checks for version numbers and returns true if latest version available
+    """
+    try:
+        # version_date variable is the current os version in the vars.tf
+        version = re.compile(r'[0-9]{8}').search(line.split("/")[-1])
+        if version:
+            version = version.group()
+            # print("version", version)
+            version_date = datetime.datetime.strptime(version, "%Y%m%d")
+            for _os in recent_os:
+                latest_gcp_os = re.compile(r'[0-9]{8}').search(_os)
+                # print(latest_gcp_os)
+                if latest_gcp_os:
+                    if "windows-server-2016" in _os and "core" not in _os and \
+                            "windows" in line.split("/")[-1]:
+                        return version_date < datetime.datetime.strptime(latest_gcp_os.group(),
+                                                                         "%Y%m%d"), _os
+                    if "centos-7" in _os and "centos" in line.split("/")[-1]:
+                        return version_date < datetime.datetime.strptime(latest_gcp_os.group(),
+                                                                         "%Y%m%d"), _os
+                    if "ubuntu-1804-bionic" in _os and "ubuntu" in line.split("/")[-1]:
+                        return version_date < datetime.datetime.strptime(latest_gcp_os.group(),
+                                                                         "%Y%m%d"), _os
+            return False, None
+        return False, None
+    except (AttributeError, TypeError) as ex:
+        print("Exception occurred: ", ex)
+
+
+def modify_file_content(file_path, recent_os):
+    """
+    Modifies os versions to latest versions available in the cloud
+    """
+    try:
+        updated = False
+        with open(file_path, 'r+') as original_file:
+            lines = original_file.readlines()
+            for index, line in enumerate(lines):
+                if "global/images/" in line:
+                    result = compare_versions(recent_os, line)
+                    if result[0]:
+                        updated = True
+                        print(f"Found old os version at line {index+1}")
+                        lines[index] = line.replace(line.split("/")[-1],
+                                                    ''.join([result[1], '"\n']))
+
+            original_file.seek(0)
+            original_file.writelines(lines)
+            if updated:
+                print("OS versions updated successfully.")
+            else:
+                print("OS versions up-to-date, no changes required.")
+            original_file.close()
+    except AttributeError as ex:
+        print("Exception occurred:", ex)
+
+
+if __name__ == "__main__":
+    # Starting point of script
+    DIR_PATH = os.path.abspath(os.path.dirname(__name__))
```

Comment:
We have given all caps, since "pylint" library is showing error. If we use normal lower case naming convention for these variables, hence these are constants, and these values are not modified later.

Reply:
Yes that's because this code exist outside of a function.  I prefer capitalized variables to be reserved for things you hard code in the file, not objects returned from function calls (since they are really not constants).  To fix this, these code should go into a function called main(), and you have 
if __name__ == "__main__":
  main()
Then pylint shouldn't complain.